Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.mm

```diff
@@ -171,6 +171,9 @@
         });
     }
 
+    if (needsFlush == NeedsFlush::Yes)
+        m_seekTo = ringBufferIndexToWrite;
+
```

Comment:
I wonder what happens if this pull code is called when m_seekTo is set but storage in the ring buffer is not done.
In that case, the pull will fail but will the next pull start where it should?

Reply:
that's the job of m_readCount.

If the data hasn't been written yet in the ring buffer, then in the pull method `m_ringBuffer->getFetchTimeBounds();` will show that the data isn't ready yet.